### PR TITLE
STYLE: Clang tidy cleanups

### PR DESCRIPTION
**Pull request by hjmjohnson**
This is an extension to #94 to modernize the codebase and bring it inline
with conventions used in ITK and VTK.

The clang-tide tool was used for most of these changes.

### CODE REVIEW

**Comment by hjmjohnson** [comment]
@dzenanz @aylward I got side tracked and cleaned this up today.

File: src/metaArrow.cxx [hunk 1]
```diff
@@ -30,17 +30,19 @@ namespace METAIO_NAMESPACE
 MetaArrow::MetaArrow()
   : MetaObject()
 {
-  if (META_DEBUG)
+  if (META_DEBUG) {
```
Comment (dzenanz) [inline src/metaArrow.cxx:33]: Commit message is misleading. These are bodies with exactly one line.

**Review by dzenanz — COMMENTED** [review]
Good refactoring, some  suggestions inline.

**Comment by hjmjohnson** [comment]
@aylward @dzenanz After addressing concerns, and updating CI to meet the new minimum CMake version requirements, this is passing all tests.

**Comment by aylward** [comment]
Looks good to me!

**Comment by aylward** [comment]
@hjmjohnson - do you want me to squash and merge - or do you have those permissions?

**Review by dzenanz — APPROVED** [review]

**Comment by dzenanz** [comment]
@aylward these commits should not be squashed. Separation of these changes into their distinct commits has significant value.

**Pull request merged**